src/Ombi.Schedule/Jobs/Emby/EmbyPlayedSync.cs

```diff
@@ -98,13 +113,117 @@ private async Task ProcessMovie(EmbyMovie movieInfo, OmbiUser user, ICollection<
+            if (recentlyAdded)
+            {
+                var recentlyAddedAmountToTake = 10; // to be adjusted?
+                episodes = await Api.GetTvPlayed(server.ApiKey, parentId, 0, recentlyAddedAmountToTake, user.ProviderUserId, server.FullUri);
+                // Setting this so we don't attempt to grab more than we need
+                if (episodes.TotalRecordCount > recentlyAddedAmountToTake)
+                {
+                    episodes.TotalRecordCount = recentlyAddedAmountToTake;
+                }
+            }
+            else
+            {
+                episodes = await Api.GetTvPlayed(server.ApiKey, parentId, 0, AmountToTake, user.ProviderUserId, server.FullUri);
+            }
+            var totalCount = episodes.TotalRecordCount;
+            var processed = 0;
+            var mediaToAdd = new HashSet<UserPlayedEpisode>();
+
+            while (processed < totalCount)
+            {
+                foreach (var episode in episodes.Items)
+                {
+                    await ProcessTv(episode, user, mediaToAdd, server);
+                    processed++;
+                }
+
+                // Get the next batch
+                // Recently Added should never be checked as the TotalRecords should equal the amount to take
+                if (!recentlyAdded)
+                {
+                    episodes = await Api.GetTvPlayed(server.ApiKey, parentId, processed, AmountToTake, user.ProviderUserId, server.FullUri);
+                }
+                await _episodeRepo.AddRange(mediaToAdd);
+                mediaToAdd.Clear();
+            }
+        }
+
+
+        private async Task ProcessTv(EmbyEpisodes episode, OmbiUser user, ICollection<UserPlayedEpisode> content, EmbyServers server)
+        {
+
+            var parent = await _contentRepo.GetByEmbyId(episode.SeriesId);
+            if (parent == null)
+            {
+                _logger.LogInformation("The episode {0} does not relate to a series, so we cannot save this",
+                    episode.Name);
+                return;
+            }
+            if (parent.TheMovieDbId.IsNullOrEmpty())
+            {
+                _logger.LogWarning($"Episode {episode.Name} is not linked to a TMDB series. Skipping.");
+                return;
+            }
+
+            await addToContent(content, new UserPlayedEpisode()
+            {
+                TheMovieDbId = int.Parse(parent.TheMovieDbId),
+                SeasonNumber = episode.ParentIndexNumber,
+                EpisodeNumber = episode.IndexNumber,
+                UserId = user.Id
+            });
+
+            if (episode.IndexNumberEnd.HasValue && episode.IndexNumberEnd.Value != episode.IndexNumber)
+            {
+                int episodeNumber = episode.IndexNumber;
+                do
+                {
+                    _logger.LogDebug($"Multiple-episode file detected. Adding episode ${episodeNumber}");
+                    episodeNumber++;
```

Comment:
This being in a `do while` loop, the first episodeNumber that get's saved is going to be incremented. Is this expected?

Reply:
The first episode has already been added a few instructions before. This will add all episodes after this one when the file spans over multiple episodes.
FYI, this is a copy and paste from the episode sync: https://github.com/sephrat/Ombi/blob/9fe7ec115e8657ed77028cbdfce1e449bfb22c1f/src/Ombi.Schedule/Jobs/Emby/EmbyEpisodeSync.cs#L171-L174